index.bs

```diff
@@ -2723,6 +2770,233 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
+
+      1. Let |node| be the result of calling [=snapshotItem=] with |evaluateResult|
+         as this and |index| as the argument.
+
+      1. Append |node| to |returned nodes|.
+      
+      1. If |maximum returned nodes| is greater than 0 and <code>length</code> of
+         |returned nodes| is greater than or equal to |maximum returned nodes|,
+         return |returned nodes|.
+
+      1. Increment |index| by 1. 
+
+1. Return |returned nodes|
+
+</div>
+
+<div algorithm="locate nodes using regexp">
+To <dfn>locate nodes using regexp</dfn> with given |context target|, |context nodes|,
+|regexp|, |maximum returned nodes|, and <var ignore>session</var>:
+
+1. Let |returned nodes| be an empty array.
+
+1. Let |element nodes| be the result of [=locate nodes using CSS=] given |context target|,
+   |context nodes|, "*" and <var ignore>session</var>.
+
+1. For each |element node| in |element nodes|:
+
+  1. Let |node text| be the <code>innerText</code> property of |element node|.
+
+  1. Let |is text match| be the result of calling [=test=] with |regexp| as
+     this and |node text| as the argument.
+
+  1. If |is text match| is true, append |element node| to |returned nodes|.
+
+  1. If |maximum returned nodes| is greater than 0 and <code>length</code> of
+     |returned nodes| is greater than or equal to |maximum returned nodes|,
+     return |returned nodes|.
+
+1. Return |returned nodes|.
+
+</div>
+
+<div algorithm="remote end steps for browsingContext.locateNodes">
+The [=remote end steps=] with |session| and |command parameters| are:
+
+1. Let |context id| be the value of the <code>context</code> field of
+   |command parameters|.
+
+1. Let |context| be the result of [=trying=] to [=get a browsing context=]
+   with |context id|.
+
+1. Assert: |context| is not null.
+
+1. Let |maximum returned nodes| be the value of the <code>maxNodes</code>
+   field of |command parameters|, if present, or 0 otherwise.
+
+1. Let |current context target| be the a [=/map=] matching of the
+   <code>script.ContextTarget</code> production with the 
+   <code>context</code> field set to the [=browsing context id=]
+   of |context|.
+
+1. Let |realm| be the result of [=trying=] to [=get a realm from a target=]
+   given |current context target|.
+
+1. Let |locator| be the value of the <code>locator</code> field
+   of |command parameters|.
+
+1. Assert: |locator| is not null.
+
+1. Let |start nodes parameter| be the value of the <code>startNodes</code> field
+   of |command parameters|.
+
+1. Assert: |start nodes parameter| is not null.
+
+1. Let |context nodes| be an empty array.
+
+1. For each |serialized start node| in |start nodes parameter|:
+
+   1. Let |start node| be the result of [=trying=] to [=deserialize shared reference=] given
+      |serialized start node|, |realm| and |session|.
+
+   1. Append |start node| to |context nodes|.
+
+1. If the <code>length</code> property of |context nodes| is 0, append the node 
```

Comment:
Don't get properties, just refer directly to the spec, in this case https://dom.spec.whatwg.org/#document-element

Also this phrasing means that passing in `startNodes: []` ends up not being an error but implicitly selecting the document element. That seems pretty non-obvious; I'd make `startNodes` optional, but require it to contain at least one element if provided.

Reply:
@jgraham Double-check my change here on how the algorithm works now. I don't know if the prose is correct. I've made `startNodes` optional, and have it's absence indicate "use the document element as the sole context node."